python/ray/_private/runtime_env/packaging.py

```diff
@@ -131,11 +142,36 @@ def parse_uri(pkg_uri: str) -> Tuple[Protocol, str]:
                 path='/dir/file.zip'
             )
             -> ("s3", "s3_bucket_dir_file.zip")
+    For HTTPS URIs, the path will have '/' and '.' replaced with '_'.
+        urlparse(
+            "https://github.com/shrekris-anyscale/test_module/archive/HEAD.zip"
+        )
+            -> ParseResult(
+                scheme='https',
+                netloc='github.com',
+                path='/shrekris-anyscale/test_repo/archive/HEAD.zip'
+            )
+            -> ("https",
+            "https_github_com__shrekris-anyscale_test_repo_archive_HEAD.zip")
+    For GS URIs, the path will have '/' replaced with '_'.
+        urlparse("gs://shreyas-runtime-env-test1/test_module/test_module.zip")
+            -> ParseResult(
+                scheme='gs',
+                netloc='shreyas-runtime-env-test1',
+                path='/test_module/test_module.zip'
+            )
+            -> ("gs",
+            "gs_shreyas-runtime-env-test1__test_module_test_module.zip")
     """
     uri = urlparse(pkg_uri)
     protocol = Protocol(uri.scheme)
-    if protocol == Protocol.S3:
-        return (protocol, f"s3_{uri.netloc}_" + "_".join(uri.path.split("/")))
+    if protocol == Protocol.S3 or protocol == Protocol.GS:
+        return (
+            protocol,
```

Comment:
We should have unit test to cover this magic url parsing and replace logic

Reply:
yep add this to test_packaging?

Reply:
I added the `test_parsing` test to `test_runtime_env_packaging.py` to cover the URL parsing and replace logic. I also cleaned the logic itself, so it relies only on `replace()` statements rather than a mix of `replace()` and `join()`.

Reply:
excellent!